### PR TITLE
release 98

### PR DESCRIPTION
**Pull request by wolfv**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

Closes https://github.com/conda-forge/binaryen-feedstock/pull/37

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   version: {{ version }}
 
 source:
-  fn: binaryen-{{ version }}.tar.gz
-  url: https://github.com/WebAssembly/binaryen/archive/{{ version }}.tar.gz
-  sha256: 4852a676c383efffa368e58f2abf3108fcf02f0e8b6cd3923b0cdc35bc0ee8c9
+  url: https://github.com/WebAssembly/binaryen/archive/version_{{ version }}.tar.gz
+  sha256: 9f805db6735869ab52cde7c0404879c90cf386888c0f587e944737550171c1c4
+  patches:
+    - gcc_stdmacro.patch  # [linux]
```
Comment (jakirkham) [inline recipe/meta.yaml:13]: Where does this patch come from? Is upstream aware of this?

File: recipe/build.sh [hunk 2]
```diff
@@ -9,15 +9,19 @@ then
...
 cmake -G "Ninja" \
-    -DCMAKE_BUILD_TYPE:STRING="${CMAKE_CONFIG}" \
-    -DCMAKE_INSTALL_PREFIX:PATH="${PREFIX}" \
-    -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=ON \
-    -DBUILD_SHARED_LIBS:BOOL=ON \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX \
+    -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+    -DBUILD_SHARED_LIBS=ON \
+    -DCMAKE_INSTALL_LIBDIR=lib \
```
Comment (jakirkham) [inline recipe/build.sh]: Is it alright if we keep the content above and just add this `LIBDIR` bit?

File: recipe/build.sh [hunk 3]
```diff
@@ -9,15 +9,19 @@ then
...
+    ${CMAKE_ARGS} \
```
Comment (jakirkham) [inline recipe/build.sh]: Where does this come from?

**Review by jakirkham — COMMENTED** [review]
Thanks Wolf! 😄 Sorry for letting things languish a bit here 😞

Had a few questions about the changes here. Generally they seem ok though. Thanks again for the work here!

File: recipe/build.sh (hunk 3 above)
Comment (wolfv) [reply recipe/build.sh]: from the CMAKE feedstock afaik. isuruf introduced this afaik.

File: recipe/build.sh (hunk 2 above)
Comment (wolfv) [reply recipe/build.sh]: sure can do.

File: recipe/meta.yaml (hunk 1 above)
Comment (wolfv) [reply recipe/meta.yaml:13]: I found this fix here (exact same issue: https://github.com/tensorflow/tensorflow/issues/12998#issuecomment-329179887)

Not sure if upstream is aware, I think it's a gcc vs clang thing.

Comment (wolfv) [reply recipe/meta.yaml:13]: Initially I attempted to build with clang, but this here is much simpler and more in line with cf now.

File: recipe/build.sh (hunk 3 above)
Comment (jakirkham) [reply recipe/build.sh]: And this exported somehow? Is conda-build passing this through somewhere? Or is it in an activation script?

**Comment by wolfv** [comment]
I reverted my changes to cmake so that it's closer to what you had. Hope this works now.

**Comment by wolfv** [comment]
@jakirkham this should be ready. It might be needed for my emscripten PR so would be cool to get this in soon!

https://github.com/conda-forge/staged-recipes/pull/13178

If you want to be maintainer on emscripten, let me know!

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml:13]: Sure using `gcc` sounds preferable. Just want to make sure this gets fixed upstream. Could you please send a PR to them?

File: recipe/gcc_stdmacro.patch [hunk 4]
```diff
@@ -0,0 +1,11 @@
+diff --git a/src/wasm/CMakeLists.txt b/src/wasm/CMakeLists.txt
+index d24618c63..28cb3aab1 100644
+--- a/src/wasm/CMakeLists.txt
++++ b/src/wasm/CMakeLists.txt
+@@ -14,4 +14,6 @@ set(wasm_SOURCES
+   wasm-validator.cpp
+   ${wasm_HEADERS}
+ )
++# needed for gcc to compile the llvm sources included here
++add_definitions(-D__STDC_FORMAT_MACROS)
```
Comment (isuruf) [inline recipe/gcc_stdmacro.patch:10]: Instead of patching, you can set it in `CXXFLAGS` and `CFLAGS` env variables

**Review by jakirkham — APPROVED** [review]

File: recipe/gcc_stdmacro.patch (hunk 4 above)
Comment (jakirkham) [reply recipe/gcc_stdmacro.patch:10]: Oops, sorry didn't see this before. Making this change also makes sense.

**Comment by jakirkham** [comment]
Thanks Wolf! 😄

**Pull request merged**